packages/docusaurus-theme-search-algolia/src/theme/SearchBar/index.js

```diff
@@ -5,135 +5,133 @@
-        <span
-          aria-label="expand searchbar"
-          role="button"
-          className={clsx(styles.searchIconButton, {
-            [styles.searchIconButtonHidden]: isSearchBarExpanded,
-          })}
-          onClick={toggleSearchInput}
-          onKeyDown={toggleSearchInput}
-          tabIndex={0}
+    <>
+      <Head>
+        {/* This hints the browser that the website will load data from Algolia,
+        and allows it to preconnect to the DocSearch cluster. It makes the first
+        query faster, especially on mobile. */}
+        <link
+          rel="preconnect"
+          href={`https://${appId}-dsn.algolia.net`}
+          crossOrigin
         />
-
-        <input
-          id="search_input_react"
-          type="search"
-          placeholder="Search"
-          aria-label="Search"
-          className={clsx('navbar__search-input', styles.searchInput, {
-            [styles.searchInputExpanded]: isSearchBarExpanded,
-          })}
-          onMouseOver={handleSearchInput}
-          onFocus={handleSearchInput}
-          onBlur={handleSearchInputBlur}
-          onKeyDown={handleSearchInputPressEnter}
-          ref={searchBarRef}
-        />
-      </div>
-    </div>
+      </Head>
+
+      <DocSearchButton
+        onTouchStart={importDocSearchModalIfNeeded}
+        onFocus={importDocSearchModalIfNeeded}
+        onMouseOver={importDocSearchModalIfNeeded}
+        onClick={onOpen}
+      />
+
+      {isOpen &&
+        createPortal(
+          <DocSearchModal
+            appId={appId}
+            apiKey={apiKey}
+            indexName={indexName}
+            searchParameters={searchParameters}
+            onClose={onClose}
+            initialScrollY={window.scrollY}
+            navigator={{
+              navigate({suggestionUrl}) {
+                history.push(suggestionUrl);
+              },
+            }}
+            transformItems={(items) => {
+              return items.map((item) => {
+                // We transform the absolute URL into a relative URL.
+                // Alternatively, we can use `new URL(item.url)` but it's not
+                // supported in IE.
+                const a = document.createElement('a');
+                a.href = item.url;
+
+                return {
+                  ...item,
+                  url: `${a.pathname}${a.hash}`,
+                };
+              });
+            }}
+            hitComponent={Hit}
+            resultsFooterComponent={(props) => (
+              <ResultsFooter {...props} onClose={onClose} />
+            )}
+          />,
+          document.body,
+        )}
+    </>
   );
-};
+}
+
+function SearchBar() {
+  const {siteConfig = {}} = useDocusaurusContext();
+
+  if (!siteConfig.themeConfig.algolia) {
+    // eslint-disable-next-line no-console
+    console.warn(`DocSearch requires an \`algolia\` field in your \`themeConfig\`.
+
+See: https://v2.docusaurus.io/docs/search/#using-algolia-docsearch`);
+
+    return null;
+  }
```

Comment:
this warning won't be needed anymore, because we have a system to validate theme config ahead of time now: https://v2.docusaurus.io/docs/next/lifecycle-apis/#validate-1

Reply:
This will be handled in another PR: https://github.com/facebook/docusaurus/pull/2815#issuecomment-661082894.